notebooks/marginalized_changepoint_model.ipynb

```diff
@@ -0,0 +1,821 @@
+{
```

Comment:
<p>Why is disasters_missing still being sampled?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymcx/pull/91/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
Those variables were not marginalized, only the switchpoint

Reply:
Also worth pointing that models with bernoulli/categoricals/shorter discrete_uniform benefit more, because the logp is not much slower than the unmarginalized model. Here the slowdown comes from having to use a Scan. I just used this example because it was already out there, and discussed among the STAN community.

Also models where you get rid of more RVs should see better benefits on the NUTS side, rather than a model with a single one like this.

Reply:
would be nice to have an example where we can use full NUTS after marginalization.

Reply:
A simple, dirty fix will be to add the missing  observations to the observed data.

One question. How difficult is to recover the marginalized parameters?

Reply:
Yes, we don't need to sample the missing years, we could also just infer/recover them in posterior predictive.

Still the original model with metropolis samples pretty well, and I see no reason why a full NUTS is going to do much better.

***

About recovering the marginalized parameters, it's not that bad. We need to sample from the marginalized elemwise logp conditioned on the remaining posterior parameters. Just some Aesara wrangling...

Reply:
I'm all for a more interesting example. Let's open an issue for that?

Reply:
Definitely should be another issue. What about a Cormack-Jolly-Seber capture-recapture model? Austin's blog post basically does what this feature could allow you to do automatically.

https://mc-stan.org/docs/stan-users-guide/mark-recapture-models.html
https://austinrochford.com/posts/2018-01-31-capture-recapture.html

Not a blocker for merging, but could be good first contribution!

Reply:
If that kind of model is supported by the current marginalization, it would be nice. Perhaps as a notebook for pymc-examples?